src/Aspire.Hosting.Azure/Extensions/AzureApplicationInsightsExtensions.cs

```diff
@@ -19,10 +24,45 @@ public static class AzureApplicationInsightsExtensions
     /// <returns>A reference to the <see cref="IResourceBuilder{AzureApplicationInsightsResource}"/>.</returns>
     public static IResourceBuilder<AzureApplicationInsightsResource> AddAzureApplicationInsights(this IDistributedApplicationBuilder builder, string name)
     {
-        var resource = new AzureApplicationInsightsResource(name);
+#pragma warning disable CA2252
+        return builder.AddAzureApplicationInsights(name, (_, _) => { });
+#pragma warning restore CA2252
+    }
+
+    /// <summary>
+    /// Adds an Azure Application Insights resource to the application model.
+    /// </summary>
+    /// <param name="builder">The builder for the distributed application.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="configureResource">Optional callback to configure the Application Insights resource.</param>
+    /// <returns></returns>
+    [RequiresPreviewFeatures]
+    public static IResourceBuilder<AzureApplicationInsightsResource> AddAzureApplicationInsights(this IDistributedApplicationBuilder builder, string name, Action<IResourceBuilder<AzureApplicationInsightsResource>, ResourceModuleConstruct>? configureResource = null)
```

Comment:
This callback should expose the application insights resource and the log analytics workspace resources from the CDK on the callback. The ability to easily get at the resources we've injected to the construct is what makes these callbacks useful.

Reply:
Woops this was a miss.